internal/reader/rmn_home.go

```diff
@@ -0,0 +1,371 @@
+	return r.sync.StopOnce(r.Name(), func() error {
+		defer r.wg.Wait()
+		close(r.stopCh)
+		return nil
+	})
+}
+
+func (r *rmnHomePoller) Ready() error {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	return r.sync.Ready()
+}
+
+func (r *rmnHomePoller) HealthReport() map[string]error {
+	r.mutex.RLock()
+	defer r.mutex.RUnlock()
+	if r.failedPolls >= MaxFailedPolls {
+		err := fmt.Errorf("polling failed %d times in a row (maximum allowed: %d)", r.failedPolls, MaxFailedPolls)
+		r.sync.SvcErrBuffer.Append(err)
+	}
+	return map[string]error{r.Name(): r.sync.Healthy()}
+}
+
+func (r *rmnHomePoller) Name() string {
+	return "rmnHomePoller"
+}
+
+func validate(config VersionedConfigWithDigest) error {
+	// check if the versionesconfigwithdigests are set (can be empty)
+	if config.ConfigDigest.IsEmpty() {
+		return fmt.Errorf("configDigest is empty")
+	}
+	return nil
+}
+
+func convertOnChainConfigToRMNHomeChainConfig(
+	lggr logger.Logger,
+	versionedConfigWithDigests []VersionedConfigWithDigest,
+) map[cciptypes.Bytes32]rmntypes.RMNHomeConfig {
+	if len(versionedConfigWithDigests) == 0 {
+		lggr.Warnw("no on chain RMNHomeConfigs found")
+		return map[cciptypes.Bytes32]rmntypes.RMNHomeConfig{}
+	}
+
+	rmnHomeConfigs := make(map[cciptypes.Bytes32]rmntypes.RMNHomeConfig)
+	for _, versionedConfigWithDigest := range versionedConfigWithDigests {
+		err := validate(versionedConfigWithDigest)
+
+		if err != nil {
+			lggr.Warnw("invalid on chain RMNHomeConfig", "err", err)
+			continue
+		}
+
+		config := versionedConfigWithDigest.VersionedConfig.Config
+		nodes := make([]rmntypes.RMNHomeNodeInfo, len(config.Nodes))
+		for i, node := range config.Nodes {
+			pubKey := ed25519.PublicKey(node.OffchainPublicKey[:])
+
+			nodes[i] = rmntypes.RMNHomeNodeInfo{
+				ID:                    rmntypes.NodeID(i),
+				PeerID:                ragep2ptypes.PeerID(node.PeerID),
+				OffchainPublicKey:     &pubKey,
+				SupportedSourceChains: mapset.NewSet[cciptypes.ChainSelector](),
+			}
+		}
+
+		minObservers := make(map[cciptypes.ChainSelector]uint64)
+
+		for _, chain := range config.SourceChains {
+			minObservers[chain.ChainSelector] = chain.MinObservers
+			for j := 0; j < len(nodes); j++ {
+				isObserver, err := IsNodeObserver(chain, j, len(nodes))
+				if err != nil {
+					lggr.Warnw("failed to check if node is observer", "err", err)
+					continue
+				}
+				if isObserver {
+					nodes[j].SupportedSourceChains.Add(chain.ChainSelector)
+				}
+			}
+		}
+
+		rmnHomeConfigs[versionedConfigWithDigest.ConfigDigest] = rmntypes.RMNHomeConfig{
+			Nodes:                   nodes,
+			SourceChainMinObservers: minObservers,
+			ConfigDigest:            versionedConfigWithDigest.ConfigDigest,
+			OffchainConfig:          config.OffchainConfig,
+		}
+	}
+	return rmnHomeConfigs
+}
+
+// IsNodeObserver checks if a node is an observer for the given source chain
+func IsNodeObserver(sourceChain SourceChain, nodeIndex int, totalNodes int) (bool, error) {
```

Comment:
This method is no doubt useful, but does it make more sense as a method on `SourceChain` maybe?